iNatImport External Link

JoeCohen opened the pull request:
- For new iNat imports
  - Adds an  ExternalLink to the iNat observation
  - Leaves inat_id `nil`
- Changes display of iNat External Links (including existing links)
From: "[On iNaturalist](https://www.inaturalist.org/observations/279505513) 2025-05-08"
To: "[iNat #279505513](https://www.inaturalist.org/observations/279505513)".

### Note:

After this is deployed, I will perform these tasks
- #2984 
- #2985

## Suggested Manual Tests
- prior manually added external link
http://localhost:3000/obs/546201
Expected external link:  [iNat #279505513](https://www.inaturalist.org/observations/279505513)
- new import
  - create a throwaway iNat obs
  - import it
  Expected external link:  `iNat #nnnnnnnnn Imported yyyy-mm-dd`
  - (Delete the throwaway iNat obs).

JoeCohen committed:
Add test (failing) for ExternalLink to iNat

JoeCohen committed:
Add ExternalLink instead of observation.inat_id

JoeCohen committed:
Change display of iNat external_link

- `On iNaturalist 2025-05-08` => `iNat #279505513 2025-05-08`

JoeCohen committed:
Include "Imported" in iNat import external link

JoeCohen committed:
Add space to display

JoeCohen committed:
Change remaining instances of observation.id

- Fixes 422 Unprocessable Entity

nimmolo commented on app/jobs/inat_import_job.rb:
```diff
@@ -42,6 +42,7 @@ def create_ivars(inat_import)
     )
     @user = @inat_import.user
     @user_api_key = APIKey.find_by(user: @user, notes: MO_API_KEY_NOTES).key
+    @external_link_site = ExternalSite.find_by(name: "iNaturalist")
```
maybe `@external_site` ?

nimmolo reviewed (approved):
Approved, but some suggestions

JoeCohen committed:
Change ivar name

Responds to https://github.com/MushroomObserver/mushroom-observer/pull/2987#discussion_r2087510569

JoeCohen committed:
Move some view logic to a helper method

Responds to @nimmolo's suggestion https://github.com/MushroomObserver/mushroom-observer/pull/2987#discussion_r2087508631

JoeCohen committed:
Merge branch 'jdc-2912-improve-inatimportjobtest' into jdc-inat-external-link

JoeCohen committed:
Simplify iNat external link

- Removes redundant "Imported" and date.
If it's imported you can tell that from the footer, and the import date is observation.created_at;
If it's not imported you can tell that from the footer, and displaying the date the link was created is weird and almost useless.
